### DISCUSSION TITLE
command line compare without user input (jenkins)

### DISCUSSION THREAD

**Discussion by angelodiego** [WinMerge/winmerge#1593] [Q&A]
I'm trying to compare two folders recursively with the following command (there is more in the command but for brevity sake I'll omit it)
"c:\program files\winmerge\WinMergeU.exe" c:\leftdir c:\rightdir -minimize -noninteractive -r -u -or c:\temp\diffreport.hml
the comparison runs in a separate process and then I get a messagge: "WinMerge: The report has been created successfully." wich expects a return to continue.
I cannot do that in my Jenkins CI, is there an undocumented parameter to wait for the end of the comparison without waiting for user input?

**Comment by sdottaka** [comment]
I may be misreading something, but with the `-noninteractive` commandline option WinMerge will not wait for user input.

If you run `"c:\program files\winmerge\WinMergeU.exe" c:\leftdir c:\rightdir -minimize -noninteractive -r -u -or c:\temp\diffreport.hml` from the command prompt , you'll find that it doesn't wait for user input.

I haven't used Jenkins very much, but I tried registering a batch file that executes the above command in Jenkins, and it seemed to work fine.

**Comment by angelodiego** [comment]
That's not the case, when you run with the -noninteractive option a new
process is spawned and after the comparison is over you get a message
telling you that the report was created, waiting for you to hit a key (abov
with the in french, notice the the prompt is not "ready" after the message)
[image: image.png]

**Reply by sdottaka** [reply]
It looks like you've attached an image to your email, but unfortunately it's not visible on GitHub.

Note that WinMerge does not have the ability to wait for keystrokes as you say.

Although it may be unrelated, in order to wait until WinMerge finishes creating the report, please execute as follows.

~~~
cmd /c start /w "" "c:\program files\winmerge\WinMergeU.exe" c:\leftdir c:\rightdir -minimize -noninteractive -r -u -or c:\temp\diffreport.hml
~~~
